ui/app/services/console.js

```diff
@@ -20,51 +20,51 @@ const VERBS = {
   delete: 'DELETE',
 };
 
-export default Service.extend({
-  isOpen: false,
+export default class ConsoleService extends Service {
+  @tracked isOpen = false;
+  @tracked commandIndex = null;
+  @tracked log = [];
 
-  adapter() {
-    return getOwner(this).lookup('adapter:console');
-  },
   get commandHistory() {
     return this.log.filter((log) => log.type === 'command');
-  },
-  log: computed(function () {
-    return [];
-  }),
-  commandIndex: null,
+  }
+
+  // Not a getter so it can be stubbed in tests
+  adapter() {
+    return getOwner(this).lookup('adapter:console');
+  }
 
   shiftCommandIndex(keyCode, setCommandFn = () => {}) {
     const [newIndex, newCommand] = shiftCommandIndex(keyCode, this.commandHistory, this.commandIndex);
     if (newCommand !== undefined && newIndex !== undefined) {
-      this.set('commandIndex', newIndex);
+      this.commandIndex = newIndex;
       setCommandFn(newCommand);
     }
-  },
+  }
 
   clearLog(clearAll = false) {
-    const log = this.log;
     let history;
     if (!clearAll) {
       history = this.commandHistory.slice();
       history.setEach('hidden', true);
     }
-    log.clear();
+    this.log = [];
     if (history) {
-      log.addObjects(history);
+      this.log = addManyToArray(this.log, history);
     }
-  },
+  }
 
   logAndOutput(command, logContent) {
-    const log = this.log;
+    const log = this.log.slice();
     if (command) {
-      log.pushObject({ type: 'command', content: command });
-      this.set('commandIndex', null);
+      log.push({ type: 'command', content: command });
     }
     if (logContent) {
-      log.pushObject(logContent);
+      log.push(logContent);
     }
-  },
+    this.log = log;
+    this.commandIndex = null;
```

Comment:
Is moving this outside the conditional intentional?

Reply:
It was, but looking at it now I'm not sure why 🤔 I'll play around with this. I have a feeling that the `commandIndex` was part of why `runCmd` test helper has been flaky

Reply:
The original intent was that the command index should only be cleared after the log is updated in the DOM (which now only happens at line 65). I think the original reason why it was only called when there's a command, is because the instances where this method is called without a command (after `api` and `refresh` are called and transition/reload the route) the index has already been cleared. So it seems harmless to keep here, but I have been wrong about harmlessness before